commands/payment_channel_daemon_test.go

```diff
@@ -1,547 +1,598 @@
+	channelAmount := types.NewAttoFILFromFIL(1000)
 
-	daemonTestWithPaymentChannel(t, &payer, &target, amt, eol, func(d *th.TestDaemon, channelID *types.ChannelID) {
-		assert := assert.New(t)
+	WithPaymentChannel(ctx, t, payerDaemon, payerAddr, targetAddr, channelAmount, channelExpiry, func(chanid *types.ChannelID) {
+		voucherAmount := types.NewAttoFILFromFIL(10)
+		voucherValidAt := types.NewBlockHeight(0)
+		voucherStr, err := payerDaemon.PaychVoucher(ctx, chanid, voucherAmount, fast.AOFromAddr(payerAddr), fast.AOValidAt(voucherValidAt))
+		require.NoError(err)
 
-		d.ConnectSuccess(targetDaemon)
+		// TODO(tperson) allow setting Price / Limit on process?
+		mcid, err := targetDaemon.PaychRedeem(ctx, voucherStr, fast.AOFromAddr(targetAddr), fast.AOPrice(big.NewFloat(0)), fast.AOLimit(300))
+		require.NoError(err)
 
-		voucher := createVoucherStr(t, d, channelID, types.NewAttoFILFromFIL(111), &payer, uint64(8))
+		_, err = targetDaemon.MessageWait(ctx, mcid)
+		require.NoError(err)
 
-		// Wait for the voucher message to be processed.
-		mustRedeemVoucher(t, targetDaemon, voucher, &target)
+		channels, err := targetDaemon.PaychLs(ctx, fast.AOFromAddr(payerAddr))
+		require.NoError(err)
 
-		ls := listChannelsAsStrs(targetDaemon, &payer)[0]
-		assert.Equal(fmt.Sprintf("%v: target: %s, amt: 10000, amt redeemed: 0, eol: 20", channelID.String(), target.String()), ls)
+		channel := channels[chanid.String()]
+		assert.Equal(channelAmount, channel.Amount)
+		assert.Equal(voucherAmount, channel.AmountRedeemed)
 	})
 }
 
-func TestPaymentChannelReclaimSuccess(t *testing.T) {
+func TestPaymentChannelRedeemTooEarlyFails(t *testing.T) {
 	t.Parallel()
 	require := require.New(t)
+	assert := assert.New(t)
 
-	// Initial Balance 10,000
-	payer, err := address.NewFromString(fixtures.TestAddresses[2])
-	require.NoError(err)
-	// Initial Balance 50,000
-	target, err := address.NewFromString(fixtures.TestAddresses[1])
-	require.NoError(err)
-
-	// Not used in logic
-	eol := types.NewBlockHeight(5)
-	amt := types.NewAttoFILFromFIL(1000)
-
-	targetDaemon := th.NewDaemon(t,
-		th.KeyFile(fixtures.KeyFilePaths()[1]),
-		// must include 0th keyfilepath if using 0th TestMiner
-		th.KeyFile(fixtures.KeyFilePaths()[0]),
-		th.WithMiner(fixtures.TestMiners[0])).Start()
-	defer targetDaemon.ShutdownSuccess()
-
-	daemonTestWithPaymentChannel(t, &payer, &target, amt, eol, func(d *th.TestDaemon, channelID *types.ChannelID) {
-		assert := assert.New(t)
-
-		d.ConnectSuccess(targetDaemon)
-
-		// payer creates a voucher to be redeemed by target (off-chain)
-		voucher := createVoucherStr(t, d, channelID, types.NewAttoFILFromFIL(10), &payer, uint64(0))
-
-		// target redeems the voucher (on-chain)
-		mustRedeemVoucher(t, targetDaemon, voucher, &target)
-
-		lsStr := listChannelsAsStrs(targetDaemon, &payer)[0]
-		assert.Equal(fmt.Sprintf("%v: target: %s, amt: 1000, amt redeemed: 10, eol: %s", channelID, target.String(), eol.String()), lsStr)
-
-		d.RunSuccess("mining once")
```

Comment:
BLOCKING: I am concerned about removing these. This pattern allowed for deterministic testing (in theory at least), now that mining is a long running process I fear that we may reintroduce test flakiness.

Reply:
See the commit prior to this one which uses the synchronous testing, as well as the top level discussion. More context in these slack discussions.

https://filecoinproject.slack.com/archives/CEHHJNJS3/p1554145557069100
https://filecoinproject.slack.com/archives/CEHHJNJS3/p1553814863008600

Reply:
https://github.com/filecoin-project/go-filecoin/issues/1873

Reply:
The commit prior to this (dcf5b35782b4dba4892588b8528a70b6b85c149a) is called (and uses) async mining. But based on this comment: https://github.com/filecoin-project/go-filecoin/pull/2319#discussion_r270228398 it sounds like we are going the synchronous route via the MiningFromCtx method, this sounds good to me (nice idea!). 

However this: https://github.com/filecoin-project/go-filecoin/pull/2319/files#r270234693 leaves me with the impression that we have not yet completed the changes required for synchronous mining. This comment is nonblocking, just trying to understand whats happening here.